### PR TITLE
[Fix][Doc] Fix LocalFile doc (#7887)

### PR DESCRIPTION
**Pull request by YOMO-Lee**
Supplement and optimize the description of the LocalFile connector on filtering files [(#7887)](https://github.com/apache/seatunnel/issues/7887)

Supplement and optimize the explanation and description of the LocalFile connector regarding file filtering

### CODE REVIEW

File: docs/en/connector-v2/source/LocalFile.md [hunk 1]
```diff
@@ -254,6 +254,12 @@ Specifies Whether to process data using the tag attribute format.
 
 Filter pattern, which used for filtering files.
 
+The filtering format is similar to wildcard matching file names in Linux.
+
+However, it should be noted that unlike Linux wildcard characters, when encountering file suffixes, the middle dot cannot be omitted.
+
+For example, `abc20241022.csv`, the normal Linux wildcard `abc*` is sufficient, but here we need to use `abc*.*` , Pay attention to a point in the middle.
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/LocalFile.md:261]:
```suggestion
Matching rules use regular expression rules.
```
I think we should tell user what rules we follow.

**Review by Hisoka-X — COMMENTED** [review]
Thanks @YOMO-Lee for raise this PR. Please modify all File series doc.

File: docs/en/connector-v2/source/LocalFile.md (hunk 1 above)
Comment (YOMO-Lee) [reply docs/en/connector-v2/source/LocalFile.md:261]: 我觉得只说正则，用户容易产生疑问，比如我，用了正则一直不生效，最好有个例子

**Review by YOMO-Lee — COMMENTED** [review]
例子在下面也有说明

File: docs/en/connector-v2/source/LocalFile.md (hunk 1 above)
Comment (Hisoka-X) [reply docs/en/connector-v2/source/LocalFile.md:261]: Yes, we can add some demo, but we should tell user what rules we follow at first. Please go ahead and add some demo. For example:
```
File Structure Example:
/project/documents/report.txt
/project/documents/notes.txt
/project/data/input.csv
/project/data/processed/output.csv
/project/data/archive/old_data.csv
/project/images/logo.png
/project/scripts/script.sh
/project/scripts/utils/helpers.sh

Matching Rules Example:
Example 1: Match all .txt files
Regular Expression:
.*/.*\.txt$

Example 2:


```

Comment (YOMO-Lee) [reply docs/en/connector-v2/source/LocalFile.md:261]: 好的，我一会完善下

**Pull request closed without merging**